atlasdb-commons/src/main/java/com/palantir/common/remoting/ServiceNotAvailableException.java

```diff
@@ -15,46 +15,78 @@
 package com.palantir.common.remoting;
 
+import com.google.common.collect.ImmutableList;
 import com.google.common.net.HostAndPort;
+import com.google.errorprone.annotations.CompileTimeConstant;
+import com.palantir.logsafe.Arg;
+import com.palantir.logsafe.Safe;
+import com.palantir.logsafe.SafeArg;
+import com.palantir.logsafe.SafeLoggable;
+import java.util.List;
+import java.util.Objects;
 import java.util.Optional;
 
 /**
- * If a server is shutting down or cannot respond to a call for another reson this exception may be thrown.
+ * If a server is shutting down or cannot respond to a call for another reason this exception may be thrown.
  * This exception indicates to the caller that this call should be retried on another server that is available.
  *
  * @author carrino
  */
-public class ServiceNotAvailableException extends RuntimeException {
+public class ServiceNotAvailableException extends RuntimeException implements SafeLoggable {
     private static final long serialVersionUID = 2L;
 
     private final Optional<HostAndPort> serviceHint;
+    private final List<Arg<?>> args;
 
-    public ServiceNotAvailableException(String message, Throwable cause, HostAndPort serviceHint) {
+    public ServiceNotAvailableException(
+            @CompileTimeConstant String message, Throwable cause, HostAndPort serviceHint, Arg<?>... args) {
         super(message, cause);
         this.serviceHint = Optional.of(serviceHint);
+        this.args = toArgListWithSafeServiceHint(args);
     }
 
-    public ServiceNotAvailableException(String message, HostAndPort serviceHint) {
+    public ServiceNotAvailableException(@CompileTimeConstant String message, HostAndPort serviceHint, Arg<?>... args) {
         super(message);
         this.serviceHint = Optional.of(serviceHint);
+        this.args = toArgListWithSafeServiceHint(args);
     }
 
-    public ServiceNotAvailableException(String message, Throwable cause) {
+    public ServiceNotAvailableException(@CompileTimeConstant String message, Throwable cause, Arg<?>... args) {
         super(message, cause);
         this.serviceHint = Optional.empty();
+        this.args = toArgListWithSafeServiceHint(args);
     }
 
-    public ServiceNotAvailableException(String message) {
+    public ServiceNotAvailableException(@CompileTimeConstant String message, Arg<?>... args) {
         super(message);
         this.serviceHint = Optional.empty();
+        this.args = toArgListWithSafeServiceHint(args);
     }
 
-    public ServiceNotAvailableException(Throwable cause) {
+    public ServiceNotAvailableException(Throwable cause, Arg<?>... args) {
         super(cause);
         this.serviceHint = Optional.empty();
+        this.args = toArgListWithSafeServiceHint(args);
     }
 
     public Optional<HostAndPort> getServiceHint() {
         return serviceHint;
     }
+
+    @Override
+    public @Safe String getLogMessage() {
+        return Objects.requireNonNullElse(getMessage(), "[No message specified]");
+    }
```

Comment:
We provide a `SafeExceptions.renderMessage` function used by the standard safeloggable exceptions. I would recommend following that pattern, otherwise diagnostic data will be missing if this code is used anywhere that isn't aware of the sls logging format (e.g. gradle plugins, tests in oss projects, etc)

Reply:
yeah - as written I think this doesn't include the args unfortunately :/